src/models/Theme.php

```diff
@@ -0,0 +1,139 @@
+use yii\helpers\Json;
+
+/**
+ * Theme Model for LUYA-Theme.
+ *
+ * @property int    $id
+ * @property string $base_path
+ * @property string $json_config
+ * @property bool   $is_active
+ *
+ * @author Bennet Klarhölter <boehsermoe@me.com>
+ * @since  3.0.0
+ */
+class Theme extends NgRestModel
+{
+    /**
+     * @inheritdoc
+     */
+    public static function tableName()
+    {
+        return 'cms_theme';
+    }
+    
+    /**
+     * @inheritdoc
+     */
+    public static function ngRestApiEndpoint()
+    {
+        return 'api-cms-theme';
+    }
+    
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [
+            [['base_path', 'json_config'], 'required'],
+            [['json_config'], 'string'],
+            [['base_path'], 'string', 'max' => 255],
+            [['is_active'], 'boolean'],
+        ];
+    }
+    
+    /**
+     * @inheritdoc
+     */
+    public function ngRestAttributeTypes()
+    {
+        return [
+            'base_path' => 'text',
+            'json_config' => ['textarea', 'encoding' => false],
+            'is_active' => ['toggleStatus', 'initValue' => 0],
+        ];
+    }
+    
+    /**
+     * @inheritdoc
+     */
+    public function ngRestScopes()
+    {
+        return [
+            ['list', ['name', 'is_active', 'base_path', 'parentTheme']],
+        ];
+    }
+    
+    public function ngRestExtraAttributeTypes()
+    {
+        return [
+            'name' => 'text',
+            'parentTheme' => 'text',
+            'author' => 'text',
+        ];
+    }
+    
+    public function ngRestActiveButtons()
+    {
+        return [
+            [
+                'class' => 'luya\admin\buttons\ToggleStatusActiveButton',
+                'attribute' => 'is_active',
+                'uniqueStatus' => true,
+                'modelNameAttribute' => 'name',
+                'label' => 'Set active',
+            ],
+        ];
+    }
+    
+    public function getName()
+    {
+        return $this->getJsonConfig('name');
+    }
+    
+    public function getParentTheme()
+    {
+        return $this->getJsonConfig('parentTheme');
+    }
+    
+    public function getAuthor()
+    {
+        return $this->getJsonConfig('author');
+    }
+    
+    private $_jsonConfig = null;
+    
+    /**
+     * Get the json config as array.
+     *
+     * @param string $node Get a given key from the config array.
+     *
+     * @return array If the given node is not found an empty array will be returned.
+     */
+    public function getJsonConfig($node = null)
+    {
+        if ($this->_jsonConfig === null) {
+            try {
+                $this->_jsonConfig = Json::decode($this->json_config);
+            } catch (InvalidArgumentException $ex) {
+                $this->_jsonConfig = [];
+            }
+        }
+        
+        if (!$node) {
+            return $this->_jsonConfig;
+        }
+        
+        if (isset($this->_jsonConfig[$node])) {
```

Comment:
maybe use short hand

```php
return isset($this->_jsonConfig[$node]) ? $this->_jsonConfig[$node] : null;
```

(keep in mind that null keys won't be found by isset and therefore i would recommend using array_key_exists instead of isset (event though the current default return value is null))